### PR TITLE
Test clean-up: Move config-related utils to libkubevirt/config

### PR DESCRIPTION
**Pull request by alromeros**
### What this PR does

This Pull Request aims to move all the config-related utils in `tests/utils.go` to the already existing `config` package. 

We are doing this as we plan to eventually remove `tests/utils.go` and we already have a package for config-related utils.

### Release note

```release-note
NONE
```

### CODE REVIEW

**Comment by alromeros** [comment]
I think it would be nice to have a follow-up PR or commit to clean up some of these functions but for the moment it's good to have these in their respective package.

**Comment by alromeros** [comment]
Note: This PR is divided into two commits: one for feature-gate-related functions and another for generic config modifications. The config helpers are interdependent, so splitting them into separate commits would be confusing. I think this division is the most consistent.

**Review by ShellyKa13 — COMMENTED** [review]
small comment which I wont insist on. But I will say that the order of the commits should be the other way around since the first commit will not run on its own.

File: tests/libkubevirt/config/featuregate.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+	return UpdateKubeVirtConfigValueAndWait(kv.Spec.Configuration)
+}
+
+func updateKubeVirtConfigValueAndWaitHandlerRedeploymnet(kvConfig v1.KubeVirtConfiguration) *v1.KubeVirt {
+	virtClient := kubevirt.Client()
+	ds, err := virtClient.AppsV1().DaemonSets(flags.KubeVirtInstallNamespace).Get(context.TODO(), "virt-handler", metav1.GetOptions{})
+	Expect(err).ToNot(HaveOccurred())
+	currentGen := ds.Status.ObservedGeneration
+	kv := testsuite.UpdateKubeVirtConfigValue(kvConfig)
+	Eventually(func() bool {
```
[the remark this answers is not in the document]
Comment (alromeros) [reply tests/libkubevirt/config/featuregate.go]: Hey @ShellyKa13 sorry for the late response. I've added `WithOffset` in the Expects and Eventuallys that lacked it. Regarding the order of the commits, both blocks use common utils (like `waitForConfigToBePropagated`) so none of them would run independently. I think that'd be better, but considering how these functions work, having each commit work independently would require more refactoring.

**Review by awels — APPROVED** [review]
@mhenriks Could you take a look as well? I can't approve.

**Comment by alromeros** [comment]
@awels @mhenriks had to rebase again, would appreciate another look, thanks!

**Pull request merged**